src/api/borrow.ts

```diff
@@ -9,12 +9,82 @@ import {
   AaveV3HydrationTestnet,
 } from "sections/lending/ui-config/addresses"
 import { useAccount } from "sections/web3-connect/Web3Connect.utils"
+import { BN_0 } from "utils/constants"
 import { H160, isEvmAccount } from "utils/evm"
 import { QUERY_KEYS } from "utils/queryKeys"
 
+export const useBorrowContractAddresses = () => {
+  const { isLoaded, evm } = useRpcProvider()
+
+  return useMemo(() => {
+    if (!isLoaded) return null
+    const isTestnet = isTestnetRpcUrl(evm.connection.url)
+    return isTestnet ? AaveV3HydrationTestnet : AaveV3HydrationMainnet
+  }, [evm, isLoaded])
+}
+
+export const useBorrowPoolDataContract = () => {
+  const { evm } = useRpcProvider()
+  const addresses = useBorrowContractAddresses()
+
+  return useMemo(() => {
+    if (!addresses) return null
+
+    return new UiPoolDataProvider({
+      uiPoolDataProviderAddress: addresses.UI_POOL_DATA_PROVIDER,
+      provider: evm,
+      chainId: parseFloat(import.meta.env.VITE_EVM_CHAIN_ID),
+    })
+  }, [addresses, evm])
+}
+
+export const useBorrowReserves = () => {
+  const { api } = useRpcProvider()
+  const poolDataContract = useBorrowPoolDataContract()
+  const addresses = useBorrowContractAddresses()
+
+  return useQuery(
+    QUERY_KEYS.borrowReserves,
```

Comment:
shouldn't query key be related to poolDataContract and addresses. So when those are changed, useBorrowReserves return different data?